connectors/sources/confluence.py

```diff
@@ -259,23 +259,31 @@ async def paginated_api_call(self, url_name, **url_kwargs):
         Yields:
             response: JSON response.
         """
-        url = os.path.join(self.host_url, URLS[url_name].format(**url_kwargs))
+        base_url = os.path.join(self.host_url, URLS[url_name].format(**url_kwargs))
+        start = 0
+
         while True:
             try:
+                url = f"{base_url}&start={start}"
                 self._logger.debug(f"Starting pagination for API endpoint {url}")
                 response = await self.api_call(url=url)
                 json_response = await response.json()
+
                 links = json_response.get("_links")
                 yield json_response
-                if links.get("next") is None:
+                if links.get("next"):
+                    url = os.path.join(
+                        self.host_url,
+                        links.get("next")[1:],
+                    )
+                elif json_response.get("start") + json_response.get("size") < json_response.get("totalSize"):
+                    start = json_response.get("start") + json_response.get("size")
+                    url = f"{base_url}&start={start}"
+                else:
                     return
-                url = os.path.join(
-                    self.host_url,
-                    links.get("next")[1:],
-                )
             except Exception as exception:
                 self._logger.warning(
-                    f"Skipping data for type {url_name} from {url}. Exception: {exception}."
+                    f"Skipping data for type {url_name} from {base_url}. Exception: {exception}."
```

Comment:
I think we'd actually still want to log out `url`, since it would let us know which "page" of results had an issue.

Reply:
Ah! Ok, hadn't caught this, but I think `url` needs to be defined outside of the loop to start. That way we can reference it here, and we also can be sure that it's not re-set on each iteration (see my most recent comment)